code/globalincs/systemvars.h

```diff
@@ -213,7 +213,56 @@ int current_detail_level();
 
 
 // Goober5000
-void insertion_sort(void *array, size_t array_size, size_t element_size, int (*fncompare)(const void *, const void *));
-
+// (Taylor says that for optimization purposes malloc/free should be used rather than vm_malloc/vm_free here)
+// NOTE: j *must* be a signed type because j reaches -1 and j+1 must be 0.  So we might as well make them ints
+// and make array_size an int as well.
+template <typename T>
+void insertion_sort(T* array_base, int array_size, int (*fncompare)(const T*, const T*))
```

Comment:
Recommend documenting the parameters, especially the `fncompare`.  I'm assuming it's formed similarly to std::operators, i.e. `bool operator>(left, right)`

Bonus points for doxygen format.

Reply:
k